app/services/search/feed_content.rb

```diff
@@ -23,11 +23,30 @@ def search_documents(params:)
       def prepare_doc(hit)
         source = hit.dig("_source")
         source["tag_list"] = hit["tags"]&.map { |t| t["name"] } || []
+        source["id"] = hit.dig("_source", "id").split("_").last.to_i
+        source["tag_list"] = hit.dig("_source", "tags")&.map { |t| t["name"] } || []
+        source["flare_tag"] = hit.dig("_source", "flare_tag_hash")
         source["user_id"] = hit.dig("_source", "user", "id")
         source["highlight"] = hit["highlight"]
+        source["readable_publish_date"] = hit.dig("_source", "readable_publish_date_string")
+
+        source.merge!(timestamps_hash(hit))
+
         source
       end
 
+      def timestamps_hash(hit)
+        if hit.dig("_source", "published_at")
+          published_at_timestamp = DateTime.parse(hit.dig("_source", "published_at")).in_time_zone
```

Comment:
I think we can use `Time.zone.parse()` for consistency (it also returns nil if given an empty string):

```ruby
[5] pry(main)> published_at = "2020-03-22T12:15:08.605Z"
=> "2020-03-22T12:15:08.605Z"
[6] pry(main)> Time.zone.parse(published_at)
=> Sun, 22 Mar 2020 12:15:08 UTC +00:00
[7] pry(main)> Time.zone.parse("")
=> nil
```

Reply:
I have an aside question though. Why are we parsing datetimes here if they are re-serialized later in the JSON sent to the frontend? Can't we leave them as strings?

```ruby
[6] pry(main)> hits.first
=> {"id"=>31,
# ....
 "published_at_int"=>1584892190,
 "published_timestamp"=>Sun, 22 Mar 2020 15:49:50 UTC +00:00}
[7] pry(main)> puts hits.first.to_json
{"id":31,
# ....
22","published_at_int":1584892190,"published_timestamp":"2020-03-22T15:49:50.603Z"}
```

Reply:
We at least have to parse it to get the integer but we can leave it for the timestamp, good call

Reply:
Can we create the `_int` field during indexing?

Reply:
_sigh_ yes that is also an option. I honestly wanted to try and let the view handle it but I couldn't figure out the js. I don't think parsing it here is a huge deal since we are agitating, yes it could be more of a problem if we were trying to export millions of these.

Reply:
Ok. I support the ideae of optimizing whenever we need to. 

BTW the JS to parse a RFC 3339/ISO 8601 string is:

```js
> const dateTimeString = "2020-03-22T12:15:08.605Z";
> const dateTime = new Date(dateTimeString);
> dateTime
Date Sun Mar 22 2020 13:15:08 GMT+0100 (Central European Standard Time)
> dateTime.getTime() // this is the number milliseconds since Epoch
1584879308605
> dateTime.getTime() / 1000
1584879308.605
> Math.floor(dateTime.getTime() / 1000)
1584879308
```

you can see it's the same value in Ruby:

```ruby
[1] pry(main)> Time.zone.parse("2020-03-22T12:15:08.605Z").to_i
=> 1584879308
```

hope this helps!

​

Reply:
Ah, thank you!!!! When I refactor to a single field I will use that!!!